lib/gateways/sessions.rb

```diff
@@ -2,16 +2,23 @@ module Gateways
   class Sessions
     MAXIMUM_RESULTS_COUNT = 500
 
-    def initialize(ips:)
-      @organisation_ips = ips
+    def initialize(ip_filter:)
+      @ip_filter = ip_filter
     end
 
     def search(username: nil, ips: nil)
-      results = Session
-        .where(siteIP: organisation_ips)
-        .where('start >= ?', 2.weeks.ago)
-        .order(start: :desc)
-        .limit(MAXIMUM_RESULTS_COUNT)
+      results = if ip_filter
```

Comment:
`methods = [[:where, 'start >= ?', 2.weeks.ago], [:order, {start: :desc}], [:limit, MAXIMUM_RESULTS_COUNT]]
      methods.prepend([:where, {siteIP: ip_filter}]) if ip_filter
      results = methods.inject(Session) { |obj, method_and_args| obj.send(*method_and_args) }`
Was going to use this instead of an ugly if statement but it became hard to read and became messy

Reply:
suggest restructuring to:

```rb
results = Session.where('start >= ?', 2.weeks.ago)...
results = results.where(siteIP: ip_filter) if !ip_filter.nil?
```

Reply:
I like